pipeline_tools/input_utils.py

```diff
@@ -0,0 +1,87 @@
+from pipeline_tools import dcp_utils
+
+
+def get_sample_id(metadata, version='4'):
+    """Return the sample id from the given metadata"""
+    if version == '4':
+        return _get_sample_id_v4(metadata)
+    else:
+        raise NotImplementedError('Only implemented for v4 metadata')
+
+
+def _get_sample_id_v4(assay_json):
+    """Return sample id from assay json"""
+    return assay_json["has_input"]
+
+
+def get_input_metadata_file_uuid(manifest_files, version='4'):
+    """Get the uuid of the file containing metadata about pipeline input files,
+    e.g. assay.json in v4"""
+    if version == '5':
+        return _get_input_metadata_file_uuid_v5(manifest_files)
+    elif version == '4':
+        return _get_input_metadata_file_uuid_v4(manifest_files)
+    else:
+        raise NotImplementedError('Only implemented for v4 and v5 metadata')
+
+
+def _get_input_metadata_file_uuid_v5(manifest_files):
+    """Get the uuid of the files.json file"""
+    return dcp_utils.get_file_uuid(manifest_files, 'files.json')
+
+
+def _get_input_metadata_file_uuid_v4(manifest_files):
+    """Get the uuid of the assay.json file"""
+    return dcp_utils.get_file_uuid(manifest_files, 'assay.json')
+
+
+def get_smart_seq_2_fastq_names(metadata, version='4'):
+    """Get the fastq file names from the given metadata"""
+    if version == '5':
+        return _get_smart_seq_2_fastq_names_v5(metadata)
+    elif version == '4':
+        return _get_smart_seq_2_fastq_names_v4(metadata)
+    else:
+        raise NotImplementedError('Only implemented for v4 and v5 metadata')
+
+
+def _get_smart_seq_2_fastq_names_v5(files_json):
```

Comment:
This looks good with the example here! https://github.com/HumanCellAtlas/metadata-schema/blob/develop/examples/JSON/v5/SmartSeq2/Q4DemoSS2Metadata_v5.json